project/OSGi.scala

```diff
@@ -0,0 +1,87 @@
+/**
+ * Copyright (C) 2016-2017 Lightbend Inc. <http://www.lightbend.com>
+ */
+package akka
+
+import com.typesafe.sbt.osgi.OsgiKeys
+import com.typesafe.sbt.osgi.SbtOsgi._
+import com.typesafe.sbt.osgi.SbtOsgi.autoImport._
+import sbt._
+import sbt.Keys._
+
+object OSGi {
+
+  // The included osgiSettings that creates bundles also publish the jar files
+  // in the .../bundles directory which makes testing locally published artifacts
+  // a pain. Create bundles but publish them to the normal .../jars directory.
+  def osgiSettings = defaultOsgiSettings ++ Seq(
+    packagedArtifact in (Compile, packageBin) := ((artifact in (Compile, packageBin)).value, OsgiKeys.bundle.value),
+    // This will fail the build instead of accidentally removing classes from the resulting artifact.
+    // Each package contained in a project MUST be known to be private or exported, if it's undecided we MUST resolve this
+    OsgiKeys.failOnUndecidedPackage := true,
+    // By default an entry is generated from module group-id, but our modules do not adhere to such package naming
+    OsgiKeys.privatePackage := Seq(),
+    // Modify constructed version to come into alignment with a valid OSGI Bundle-Version
+    OsgiKeys.bundleVersion := version.value.replaceFirst("\\+", ".").replaceAll("\\+", "-")
+  )
+
+  // Need to specify scala.reflect.macros here because the bnd macro will not import this with the proper version
+  val parsing = osgiPkgHeaders(
+    exports = Seq("akka.parboiled2.*", "akka.shapeless.*", "akka.macros.*"),
+    imports = Seq("scala.reflect.macros", optionalResolution("scala.quasiquotes")))
+
+  val httpCore = osgiPkgHeaders(
+    exports = Seq("akka.http.*"),
+    imports = Seq(scalaJava8CompatImport, akkaParboiledImport, akkaShapelessImport, akkaMacrosImport))
+
+  val http = osgiPkgHeaders(
+    exports = Seq(
+      "akka.http.scaladsl.client.*",
+      "akka.http.$DSL$.server.*",
```

Comment:
Can we pick a different char than `$` here? This will cause scala compiler warnings that *"hey, did you forget to say `s"$DSL"`?"* Let's go with `#` maybe